### PR TITLE
Use a lock & a timeout when checking whether an upload target file is open.

### PR DESCRIPTION
**Pull request by baconmania**

### CODE REVIEW

File: SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityUploader.java [hunk 1]
```diff
@@ -68,6 +72,8 @@
     this.logIdentifier = String.format("[%s]", metadataPath.getFileName());
     this.configuration = configuration;
     this.exceptionNotifier = exceptionNotifier;
+
+    this.checkFileOpenLock = new ReentrantLock();
```
Comment (ssalinas) [inline SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityUploader.java]: Just checking, is this class always a singleton? Want to make sure we aren't just creating multiple locks. I had thought the singleton was the uploader driver, not the uploader

Comment (baconmania) [reply SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityUploader.java]: Good catch. Fixed.

**Comment by ssalinas** [comment]
🚢

**Comment by ssalinas** [comment]
🚢

**Pull request merged**